pkg/test/tmpl/parameters.go

```diff
@@ -0,0 +1,24 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package tmpl
+
+// Parameters is the set of standard parameters that can bs used by templates within tests.
+type Parameters struct {
+	// The namespace of the cluster used by the tests.
+	TestNamespace string
```

Comment:
If these are truly environment agnostic parameters, should this just be a property of the `Environment` interface?

Reply:
There is a tension here. If we try to make this environment agnostic, then the parameters needs to come from environment (somehow). That is a more complicated code pattern. It is simpler to have a union of all possible variables across environments.

Reply:
Yup, understood...  It's fine for now.